src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
+  });
+});
+
+let ASC = true;
+let currentHeader;
+
+// Sort table by column
+headers.forEach((header, i) => {
+  header.addEventListener('click', (e) => {
+    ASC = !ASC;
+
+    if (currentHeader !== i) {
+      ASC = true;
+    }
+    currentHeader = i;
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      const checkNumberCellA = strToNumber(cellA);
+      const checkNumberCellB = strToNumber(cellB);
+
+      if (Number(checkNumberCellA)) {
+        return ASC
+          ? checkNumberCellA - checkNumberCellB
+          : checkNumberCellB - checkNumberCellA;
+      }
+
+      return ASC ? cellA.localeCompare(cellB) : cellB.localeCompare(cellA);
+    });
+
+    rows.forEach((row) => tbody.appendChild(row));
+  });
+});
+
+const formFields = {
+  name: {
+    label: 'Name',
+    name: 'name',
+    type: 'text',
+    'data-qa': 'name',
+    validation: {
+      minLength: 4,
+    },
+  },
+  position: {
+    label: 'Position',
+    name: 'position',
+    type: 'text',
+    'data-qa': 'position',
+  },
+  office: {
+    label: 'Office',
+    name: 'office',
+    'data-qa': 'office',
+    options: [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ],
+  },
+  age: {
+    label: 'Age',
+    name: 'age',
+    type: 'number',
+    'data-qa': 'age',
+    validation: {
+      min: 18,
+      max: 90,
+    },
+  },
+  salary: {
+    label: 'Salary',
+    name: 'salary',
+    type: 'number',
+    'data-qa': 'salary',
+  },
+  startDate: '',
+};
+
+const strToNumber = (n) => Number(n.replace(/[^0-9.-]+/g, ''));
+
+// Add new employee form
+
+const createForm = (fields) => {
+  const formEl = document.createElement('form');
+
+  formEl.classList.add('new-employee-form');
+
+  for (const key in fields) {
+    if (fields[key]) {
+      const label = document.createElement('label');
+
+      label.textContent = `${fields[key].label}: `;
+
+      let field = document.createElement('input');
+
+      if (fields[key].hasOwnProperty('options')) {
+        field = document.createElement('select');
+
+        for (const option of fields[key].options) {
+          const optionEl = document.createElement('option');
+
+          optionEl.value = option;
+          optionEl.textContent = option;
+          field.appendChild(optionEl);
+        }
+      }
+
+      label.appendChild(field);
+
+      field.setAttribute('name', fields[key].name);
+      field.setAttribute('type', fields[key].type);
+      field.setAttribute('data-qa', fields[key]['data-qa']);
+      field.setAttribute('min', fields[key].validation?.min);
+      field.setAttribute('max', fields[key].validation?.max);
+      field.setAttribute('minlength', fields[key].validation?.minLength);
+      field.setAttribute('required', true);
+      formEl.appendChild(label);
+    }
+  }
+
+  return formEl;
```

Comment:
This function `createForm` is responsible for creating the form. It would be more efficient and readable to separate this logic into smaller functions, for example, a function that creates a label, a function that creates an input field, a function that creates a select field, and a function that creates an option. This would make the code more maintainable and testable.